androidApp/src/androidMain/kotlin/org/jetbrains/kotlinconf/android/MainActivity.kt

```diff
@@ -27,14 +28,16 @@ class MainActivity : ComponentActivity() {
 
         processIntent(intent)
 
-        val platformModule = platformModule(
-            activity = this,
-            application = application,
-            notificationIconId = R.drawable.kotlinconf_notification_icon,
-        )
+        val permissionHandler = PermissionHandler(activity = this)
+        val activityModule = module {
+            single<PermissionHandler> { permissionHandler }
+        }
+        KoinPlatformTools.defaultContext()
+            .get()
+            .loadModules(listOf(activityModule))
```

Comment:
instead of loading a side module here, you can have a definition with dynamic param like:
`factory { (activity : ComponentActivity) -> PermissionHandler(activity = activity) }`

When injecting, you pass the param. In MainActivity, you can have a field: `val permissionHandler : PermissionHandler by inject { parametersOf(this)}`. This will pass the activity as parameter.

Reply:
Unfortunately we inject the `PermissionHandler` in a deeper layer of the application, inside the `NotificationService` class. This is a data layer dependency of a `ViewModel`, and down there we have no way of accessing the `Activity` reference for injection.

Reply:
then in this case, you can declare on the fly your instance. It avoid you to have to define a module.
In the activity, you can do `getKoin().declare(PermissionHandler(activity = this))`

And later inject your PermissionHandler.

Reply:
Neat, changed this to use `declare`!